tests: Bluetooth: tester: Add to CI native_posix build targets

MariuszSkamra opened the pull request:
This extends the list of build variants with native_posix variant to build the tester application by CI.

MariuszSkamra committed:
tests: Bluetooth: tester: Add native_posix board to test build targets

This extends the list of build variants with native_posix variant to
build the tester application by CI.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

Thalley commented on tests/bluetooth/tester/testcase.yaml:
```diff
@@ -14,6 +20,13 @@ tests:
+    build_only: true
+    platform_allow:
+      - native_posix
+    extra_args: OVERLAY_CONFIG="overlay-le-audio.conf;overlay-native.conf"
```
What is the reason why we have a `overlay_native` rather than a `boards/native_posix.conf`? 

If we had the above, we should be able to just add `native_posix` to the `platform_allow` above, right?

MariuszSkamra replied:
I've tried to do it exactly as you suggested, but west build does not apply the native_posix.conf for some reason.

Thalley replied:
> I've tried to do it exactly as you suggested, but west build does not apply the native_posix.conf for some reason.

Hmm, strange. IIRC there is something about CI only picking and building for 1 of the possible targets, and not all (for each CI run). Maybe that's the case?

MariuszSkamra replied:
I figured out what was the reason. The overlays and configurations were not located under ./board directory so that were not applied by build system

MariuszSkamra committed:
Bluetooth: mesh: Fix uninitialized variable error

This fixes the error below:

main.c: In function ‘bt_mesh_provision’:
main.c:173:17: error: ‘node’ may be used uninitialized in
	       this function [-Werror=maybe-uninitialized]
  173 |                 bt_mesh_cdb_node_del(node, true);
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

MariuszSkamra committed:
tests: Bluetooth: tester: Move DT overlays to boards directory

This moves the board DT overlays and configurations to /board
directory so that it can be automatically applied in compilation
pipeline.

Signed-off-by: Mariusz Skamra <mariusz.skamra@codecoup.pl>

PavelVPV commented on subsys/bluetooth/mesh/main.c:
```diff
@@ -169,7 +167,7 @@ int bt_mesh_provision(const uint8_t net_key[16], uint16_t net_idx,
 
 end:
-	if (err && is_cdb_dev_key_valid && IS_ENABLED(CONFIG_BT_MESH_CDB)) {
+	if (err && node != NULL) {
```
```suggestion
	if (IS_ENABLED(CONFIG_BT_MESH_CDB) && err && node != NULL) {
```
`bt_mesh_cdb_node_del` is only compiled if `CONFIG_BT_MESH_CDB` is enabled.

MariuszSkamra replied:
The code will be compiled out anyway, as `node` is always NULL if `CONFIG_BT_MESH_CDB` is disabled

PavelVPV replied:
When compiling with `CONFIG_NO_OPTIMIZATIONS`:

> [203/203] Linking C executable zephyr/zephyr.elf
...
/usr/bin/ld.bfd: app/libapp.a(main.c.obj): warning: relocation in read-only section `.rodata.comp'
/usr/bin/ld.bfd: zephyr/subsys/bluetooth/mesh/libsubsys__bluetooth__mesh.a(main.c.obj): in function `bt_mesh_provision':
/home/pavel/nordic/ncs/git-repos/zephyrproject/zephyr/subsys/bluetooth/mesh/main.c:171: undefined reference to `bt_mesh_cdb_node_del'

MariuszSkamra replied:
done

Thalley commented on tests/bluetooth/tester/boards/nrf52840dk_nrf52840.overlay:
When these files exist in the `boards` directory, should they be renamed to `.conf` instead of `.overlay`?

MariuszSkamra replied:
nope, `.conf` is for board configuration overlay while `.overlay` is for DT overlay

Thalley replied:
But the board files work as an overlay on the `prj.conf` file, right? At least that's how they are used most places where the have board-specific configurations file. 

Either way, if it works, it works :D

Thalley commented on tests/bluetooth/tester/testcase.yaml:
```diff
@@ -18,13 +20,8 @@ tests:
     build_only: true
     platform_allow:
       - qemu_x86
+      - native_posix
```
Did you get it to work?

MariuszSkamra replied:
yes

MariuszSkamra replied:
All tests passed :)
